### PR TITLE
Added support for datetime types and related tests

### PR DESCRIPTION
**Pull request by AureliaRoumesy**
Added support for all `datetime` python types https://docs.python.org/fr/3/library/datetime.html#available-types

### CODE REVIEW

File: py_ts_interfaces/parser.py [hunk 1]
```diff
@@ -19,6 +19,12 @@ class PossibleInterfaceReference(str):
     "int": "number",
     "float": "number",
     "complex": "number",
+    "datetime": "Date",
+    "date": "Date",
+    "time": "Date",
+    "timedelta": "Date",
```
Comment (adrienthiery) [inline py_ts_interfaces/parser.py]: Should a `timedelta` really be a date ? 🤔 
image

File: py_ts_interfaces/parser.py [hunk 2]
```diff
@@ -19,6 +19,12 @@ class PossibleInterfaceReference(str):
     "int": "number",
     "float": "number",
     "complex": "number",
+    "datetime": "Date",
+    "date": "Date",
+    "time": "Date",
```
Comment (adrienthiery) [inline py_ts_interfaces/parser.py]: I don't think `Date` is a good equivalent for a `time` type 🤔 
The closest I see that is standardized might be the `<input type="time" />`, which would be a string with a specific format?
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/time

File: py_ts_interfaces/parser.py [hunk 3]
```diff
@@ -19,6 +19,12 @@ class PossibleInterfaceReference(str):
...
+    "timedelta": "Date",
+    "tzinfo": "Date",
+    "timezone": "Date",
```
Comment (adrienthiery) [inline py_ts_interfaces/parser.py]: Same for timezone types, I don't think `Date` is the appropriate TS equivalent. Probably more of a string

File: README.md [hunk 4]
```diff
@@ -241,6 +245,47 @@ export interface Simple2 extends Simple0, Simple1 {
...
+## Support for datetime and date
+
+Support for `datetime` and `date` types is available through `-dt, --date-type` flag. We have decided to add this support in a configurable way as dates can be transformed in different types depending the projet / implementation. Possible choices are `["string", "number", "Date"]` and the default one is `string` (because it's the return type of a `Date.prototype.toJSON()`).
+
+Examples with below python class:
+
+```python
+from dataclasses import dataclass
+from datetime import datetime, date
+# datetime and date have to be imported this way and not with:
```
Comment (mp-freatic) [inline README.md:257]: We should be able to use both, no?

File: python_to_typescript_interfaces/cli.py [hunk 5]
```diff
@@ -54,6 +57,16 @@ def get_args_namespace() -> argparse.Namespace:
         dest="should_export",
         help="whether the interface definitions should be prepended with `export`",
     )
+    argparser.add_argument(
+        "-dt, --date-type",
+        type=str,
+        dest="date_transformed_type",
+        help="defined the type dates should be tranformed in, (default: %(default)s)",
```
Comment (mp-freatic) [inline python_to_typescript_interfaces/cli.py:64]: `defines` or `define` ?
`transformed to` and not `transformed in,`

File: python_to_typescript_interfaces/cli.py [hunk 6]
```diff
@@ -54,6 +57,16 @@ def get_args_namespace() -> argparse.Namespace:
...
+        choices=["string", "number", "Date"],
+        default="string",
+        const="string",
+        nargs="?",
```
Comment (mp-freatic) [inline python_to_typescript_interfaces/cli.py:68]: Is that obligatory?

File: python_to_typescript_interfaces/parser.py [hunk 7]
```diff
@@ -29,6 +29,11 @@ class PossibleInterfaceReference(str):
     "Union": "any",
 }
 
+SUPPORTED_DATE_TYPES: List[str] = [
+    "datetime",
+    "date",
```
Comment (mp-freatic) [inline python_to_typescript_interfaces/parser.py:34]: You could simply add `datetime.datetime` and `datetime.date` in here, no?

**Review by adrienthiery — APPROVED** [review]

**Pull request merged**